receiver/kubeletstatsreceiver/internal/kubelet/metadata.go

```diff
@@ -50,15 +51,98 @@ type Metadata struct {
 	Labels                    map[MetadataLabel]bool
 	PodsMetadata              *v1.PodList
 	DetailedPVCResourceSetter func(rb *metadata.ResourceBuilder, volCacheID, volumeClaim, namespace string) error
+	PodResources              map[string]resources
+	ContainerResources        map[string]resources
+}
+
+type resources struct {
+	cpuLimit      float64
+	cpuRequest    float64
+	memoryLimit   float64
+	memoryRequest float64
+}
+
+func getContainerResources(r *v1.ResourceRequirements) resources {
+	if r == nil {
+		return resources{}
+	}
+	var containerResource resources
+	cpuLimit, err := strconv.ParseFloat(r.Limits.Cpu().AsDec().String(), 64)
+	if err == nil {
+		containerResource.cpuLimit = cpuLimit
+	}
+	cpuRequest, err := strconv.ParseFloat(r.Requests.Cpu().AsDec().String(), 64)
+	if err == nil {
+		containerResource.cpuRequest = cpuRequest
+	}
+	memoryLimit, err := strconv.ParseFloat(r.Limits.Memory().AsDec().String(), 64)
```

Comment:
Why parsing from string? `r.Requests.Memory().Value()` should give us precise value in bytes